### ISSUE TITLE
2.21.1: XDG_CONFIG_DIR -> XDG_DATA_HOME migrate regression

### ISSUE THREAD

**Issue by glensc** [direnv/direnv#574]
My direnv was upgraded via brew to 2.21.1 (I don't know from what version)

Seems the migration has failed:

```
➔ cd ../project
direnv: error.envrc is blocked. Run `direnv allow` to approve its content
➔ direnv allow

Migrating the allow data to the new location

The allowed .envrc permissions used to be stored in the XDG_CONFIG_DIR. It's
better to keep that folder for user-editable configuration so the data is
being moved to XDG_DATA_HOME.

moving /Users/glen/.config/direnv/allow to /Users/glen/.local/share/direnv/allow
direnv: error rename /Users/glen/.config/direnv/allow /Users/glen/.local/share/direnv/allow: no such file or directory
➔ direnv --version
2.21.1
➔
```

```
➔ ls -ld /Users/glen/.config/direnv/allow /Users/glen/.local/share/direnv/allow /Users/glen/.local/share/direnv /Users/glen/.local/share
ls: cannot access '/Users/glen/.local/share/direnv/allow': No such file or directory
ls: cannot access '/Users/glen/.local/share/direnv': No such file or directory
drwxr-xr-x 77 glen staff 2464 jaan  31 14:31 /Users/glen/.config/direnv/allow/
drwx------  8 glen staff  256 dets  12 13:01 /Users/glen/.local/share/
➔
```

**Comment by glensc** [comment]
I've fixed problem locally by making parent dirs:

```
➔ mkdir -p /Users/glen/.local/share/direnv
➔ direnv allow

Migrating the allow data to the new location

The allowed .envrc permissions used to be stored in the XDG_CONFIG_DIR. It's
better to keep that folder for user-editable configuration so the data is
being moved to XDG_DATA_HOME.

moving /Users/glen/.config/direnv/allow to /Users/glen/.local/share/direnv/allow
creating a symlink back from /Users/glen/.local/share/direnv/allow to /Users/glen/.config/direnv/allow for back-compat.

All done, have a nice day!
direnv: loading ~/scm/delfi/dgs/dgs-admin/.envrc
direnv: loading .envrc.local
direnv: export +GITLAB_API_ENDPOINT +GITLAB_API_PRIVATE_TOKEN +K8S_APP +K8S_NAMESPACE +KUBE_PS1_LAST_TIME +PHP_VERSION ~PATH
➔
```

I'm surprised direnv didn't consider that parent dirs can be missing when moving directories.

**Comment by zimbatm** [comment]
this has been fixed in 2.21.2

**Comment by glensc** [comment]
duplicate: https://github.com/direnv/direnv/issues/565

updated in brew: https://github.com/Homebrew/homebrew-core/pull/49497
